tjpcov/nmt_tools.py

```diff
@@ -0,0 +1,438 @@
+    -----------
+        sacc_data (Sacc):  Data Sacc instance
+        tracer (str):  Tracer name
+
+    Returns:
+    --------
+        spin (int):  Spin of the given tracer
+
+    """
+    tr = sacc_data.get_tracer(tracer)
+    if tr.quantity in ['cmb_convergence', 'galaxy_density']:
+        return 0
+    elif tr.quantity == 'galaxy_shear':
+        return 2
+
+
+def get_tracer_comb_spin(sacc_data, tracer_comb):
+    """
+    Return the spins of a pair of tracers
+
+    Parameters:
+    -----------
+        sacc_data (Sacc):  Data Sacc instance
+        tracer_comb (tuple):  List or tuple of a pair of tracer names
+
+    Returns:
+    --------
+        s1 (int):  Spin of the first tracer
+        s2 (int):  Spin of the second tracer
+
+    """
+    s1 = get_tracer_spin(sacc_data, tracer_comb[0])
+    s2 = get_tracer_spin(sacc_data, tracer_comb[1])
+
+    return s1, s2
+
+
+def get_tracer_comb_dof(sacc_data, tracer_comb):
+    """
+    Return the degrees of freedom of the Cell of a pair of tracers
+
+    Parameters:
+    -----------
+        sacc_data (Sacc):  Data Sacc instance
+        tracer_comb (tuple):  List or tuple of a pair of tracer names
+
+    Returns:
+    --------
+        dof (int):  Degrees of freedom of the Cell of the pair of tracers
+        given
+
+    """
+    dof1 = get_tracer_dof(sacc_data, tracer_comb[0])
+    dof2 = get_tracer_dof(sacc_data, tracer_comb[1])
+
+    return dof1 * dof2
+
+
+def get_datatypes_from_dof(dof):
+    """
+    Return the possible datatypes (cl_00, cl_0e, cl_0b, etc.) given a number
+    of degrees of freedom
+
+    Parameters:
+    -----------
+        dof (int):  Degrees of freedom of the Cell of the pair of tracers
+        given
+
+    Returns:
+    --------
+        datatypes (list):  List of data types assotiated to the given degrees
+        of freedom
+
+    """
+    # Copied from https://github.com/xC-ell/xCell/blob/069c42389f56dfff3a209eef4d05175707c98744/xcell/cls/to_sacc.py#L202-L212
+    if dof == 1:
+        cl_types = ['cl_00']
+    elif dof == 2:
+        cl_types = ['cl_0e', 'cl_0b']
+    elif dof == 4:
+        cl_types = ['cl_ee', 'cl_eb', 'cl_be', 'cl_bb']
+    else:
+        raise ValueError('dof does not match 1, 2, or 4.')
+
+    return cl_types
+
+
+def get_cl_for_cov(clab, nlab_cp, ma, mb, w):
+    """
+    Computes the coupled Cell that goes into the covariance matrix
+
+    Parameters:
+    -----------
+        clab (array): Fiducial Cell for the tracers a and b, used for in the
+        covariance estimation
+        nlab_cp (array): Coupled noise for the tracers a and b
+        ma (array): Mask of the field a
+        mb (array): Mask of the field b
+        w (NmtWorkspace): NmtWorkspace of the fields a and b
+
+    Returns:
+    --------
+        cl:  Coupled Cell with signal and noise
+
+    """
+    mean_mamb = np.mean(ma * mb)
+    if not mean_mamb:
+        cl_cp = np.zeros_like(nlab_cp)
+    else:
+        cl_cp = (w.couple_cell(clab) + nlab_cp) / mean_mamb
```

Comment:
That is true. However, when the tracers have no overlap `mean_mamb = 0` and `cl_cp = np.zeros_like(nlab_cp)`

Reply:
Related to this, in https://github.com/LSSTDESC/TJPCov/pull/21/commits/b2a626dd711fff3c4117e2c9a2616afbcceccaca I have modified `get_workspaces_dict` to check if `mean_mamb = 0` to avoid computing the workspace in that case.